6.3.long_cat_is_long.py

```diff
@@ -0,0 +1,21 @@
+def count_words(text):
+    """
+    Counts the length of each word in a given text and returns a dictionary of the lengths and their frequencies.
+    :param text: The text to count the word lengths from.
+    :return: A dictionary of the lengths of each word and their frequencies.
+    """
+    # Clean text from non-alphabetic characters using a generator expression
+    words = (word.strip().strip('.,!?') for word in text.split())
+    # Use a dictionary comprehension to count the lengths of the words
+    word_lengths = {word: 0 for word in words}
+    for word in text.split():
+        # Strip word from non-alphabetic characters
+        word = word.strip().strip('.,!?')
+        if word in word_lengths:
+            word_lengths[word] += 1
```

Comment:
It seem you didn't understand what was asked.
we want the number to be the length of the word, not how many times it appears in the text..